### ISSUE TITLE
Consider to add or replace an alternative octagon shape definition

### ISSUE THREAD

**Issue by rdemaria** [MethodicalAcceleratorDesign/MAD-X#1056]
The present OCTAGON is defined through 2 coordinates and 2 angles.
The angles are normally computed by the user and not directly available in technical drawings.
It could be interesting to have a specification that is closer to quantities to be extract from technical drawings.

**Comment by rdemaria** [comment]
One option could be using 

|x|<aper1
|y|<aper2
|x-y|<aper3
|x+y|<aper4

**Comment by ldeniau** [comment]
I suggest using what I did in MAD-NG, which is both simple and flexible to define, numerically stable, and fast to check:
octagon: A rectangular shape with corners truncated by a triangular shape with four parameters defining the x,y-half lengths and the triangle x,y-side lengths. **An octagon can model hexagon or diamond shapes** by equating the triangle lengths to the rectangle half-lengths.

The code for checking is rather simple and can be found here: 
https://github.com/MethodicalAcceleratorDesign/MAD/blob/c0baf2c2cf45e1a5430fcc69d73e1e5eab40d8c3/src/madl_aper.mad#L56

**Comment by rdemaria** [comment]
The most common case is defined by the cut in H,V and 45 degree. 

I would use these 3 values as ap1, ap2 and ap3. We could use ap4 for a generalization but without changing the meaning of ap1, 2, 3 or just not to define it at all.

**Comment by ldeniau** [comment]
45 degree means ap3 == ap4, so it's easy to say ap4 = ap3 if not present.

**Comment by ldeniau** [comment]
the octagon check become:
```
  octagon    = \x,y,ap -> x < ap[1] and y < ap[2] and
                          y < (x-(ap[1]-ap[3]))*(ap[4] or ap[3])/ap[3] + ap[2],
```

**Comment by rdemaria** [comment]
yes I agree. The definition in madng is interesting, however in practice I would need to multiply the aperture at 45* by sqrt(2) to get ap[3]. It is not very practical.

I would propose

|x-y|<aper3*sqrt(2)
|x+y|<aper3*sqrt(2)

an aper4 does not appear immediately useful to me. I think it is better to wait and define if an application will require it and adapt to it in case.

**Comment by ldeniau** [comment]
For a regular octagon, you need only ap1, and for non-regular octagon, you need the ap1...ap4, so I don't understand your point about ap4...
[Untitled (dragged).pdf](https://github.com/MethodicalAcceleratorDesign/MAD-X/files/7610606/Untitled.dragged.pdf)
.

**Comment by ldeniau** [comment]
As a side remark about the flexibility of the octagon, we get for free and cheap calculation:
- ap1=ap3 *or* ap2=ap4 gives a hexagon (that we don't have)
- ap1=ap3 *and* ap2=ap4 gives a diamond (that we don't have)
